SessionShareExtension/SAEScreenLockViewController.swift

```diff
@@ -0,0 +1,206 @@
+// Copyright © 2022 Rangeproof Pty Ltd. All rights reserved.
+
+import UIKit
+import PromiseKit
+import SignalCoreKit
+import SignalUtilitiesKit
+import SessionUIKit
+import SessionUtilitiesKit
+
+final class SAEScreenLockViewController: ScreenLockViewController, ScreenLockViewDelegate {
+    private var hasShownAuthUIOnce: Bool = false
+    private var isShowingAuthUI: Bool = false
+    
+    private weak var shareViewDelegate: ShareViewDelegate?
+    
+    // MARK: - Initialization
+    
+    init(shareViewDelegate: ShareViewDelegate) {
+        super.init(nibName: nil, bundle: nil)
+        
+        self.shareViewDelegate = shareViewDelegate
+        self.delegate = self
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    deinit {
+        OWSLogger.verbose("Dealloc: \(type(of: self))")
+    }
+    
+    // MARK: - UI
+    
+    private lazy var gradientBackground: CAGradientLayer = {
+        let layer: CAGradientLayer = CAGradientLayer()
+        
+        let gradientStartColor: UIColor = (LKAppModeUtilities.isLightMode ?
+            UIColor(rgbHex: 0xFCFCFC) :
+            UIColor(rgbHex: 0x171717)
+        )
+        let gradientEndColor: UIColor = (LKAppModeUtilities.isLightMode ?
+            UIColor(rgbHex: 0xFFFFFF) :
+            UIColor(rgbHex: 0x121212)
+        )
+        layer.colors = [gradientStartColor.cgColor, gradientEndColor.cgColor]
+        
```

Comment:
We have a style guide in SessionUIKit. Could you see if you can move these values to `Colors`?

Reply:
@RyanRory This was just the refactor from ObjC into Swift so these were the existing colours (they match the ones found https://github.com/oxen-io/session-ios/blob/master/SignalUtilitiesKit/Screen%20Lock/ScreenLockViewController.m#L45) - Do you happen to know which of these screens is meant to have the gradient? (it doesn't seem like it's working anyway)

<img width="200" alt="Screen Shot 2022-01-14 at 1 07 45 pm" src="https://user-images.githubusercontent.com/15862619/149439167-fb14a62e-bddc-4c5a-b99f-7cd72ca24760.png"><img width="200" alt="Screen Shot 2022-01-14 at 1 07 38 pm" src="https://user-images.githubusercontent.com/15862619/149439175-3d154586-fdd6-4a77-b8fa-88aff37a1ab3.png">

Taking a look this screen actually seems a bit busted, and looks like it is in the current version as well...
<img width="200" alt="Screen Shot 2022-01-14 at 1 08 53 pm" src="https://user-images.githubusercontent.com/15862619/149439303-697cc373-47ea-4adf-a371-a5ab87789e7f.png">